lib/gdal/extensions/dataset/extensions.rb

```diff
@@ -0,0 +1,371 @@
+          options = { pixel_value_field: pixel_value_field }
+          options[:mask_band] = band.mask_band if use_band_masks
+          band.polygonize(layer, **options)
+        end
+
+        if block_given?
+          yield data_source
+          data_source.close
+        end
+
+        data_source
+      end
+
+      # Gets the OGR::Geometry that represents the extent of the dataset.
+      #
+      # @return [OGR::Polygon]
+      # TODO: This should return an OGR::Envelope.
+      def extent
+        ul = geo_transform.apply_geo_transform(0, 0)
+        ur = geo_transform.apply_geo_transform(raster_x_size, 0)
+        lr = geo_transform.apply_geo_transform(raster_x_size, raster_y_size)
+        ll = geo_transform.apply_geo_transform(0, raster_y_size)
+
+        ring = OGR::LinearRing.new
+        ring.point_count = 5
+        ring.set_point(0, ul[:x_geo], ul[:y_geo])
+        ring.set_point(1, ur[:x_geo], ur[:y_geo])
+        ring.set_point(2, lr[:x_geo], lr[:y_geo])
+        ring.set_point(3, ll[:x_geo], ll[:y_geo])
+        ring.set_point(4, ul[:x_geo], ul[:y_geo])
+
+        polygon = OGR::Polygon.new
+        polygon.add_geometry(ring)
+        polygon.spatial_reference = spatial_reference
+
+        polygon
+      end
+
+      # @param wkt_geometry_string [String]
+      # @param wkt_srid [Integer]
+      # @return [Boolean]
+      def contains_geometry?(wkt_geometry_string, wkt_srid = 4326)
+        source_srs = OGR::SpatialReference.new_from_epsg(wkt_srid)
+        source_geometry = OGR::Geometry.create_from_wkt(wkt_geometry_string, source_srs)
+        @raster_geometry ||= extent
+
+        coordinate_transformation = OGR::CoordinateTransformation.new(source_srs,
+                                                                      @raster_geometry.spatial_reference)
+        source_geometry.transform!(coordinate_transformation)
+
+        @raster_geometry.contains? source_geometry
+      end
+
+      # Retrieves pixels from each raster band and converts this to an array of
+      # points per pixel.  For example:
+      #
+      #   # If the arrays for each band look like:
+      #   red_band_array = [0, 0, 0]
+      #   green_band_array = [10, 10, 10]
+      #   blue_band_array = [99, 99, 99]
+      #   alpha_band_array = [250, 150, 2]
+      #
+      #   # This array would look like:
+      #   [[0, 10, 99, 2], [0, 10, 99, 150], [0, 10, 99, 250]]
+      # @return NArray
+      def to_na(to_data_type = nil)
+        na = NMatrix.to_na(raster_bands.map { |r| r.to_na(to_data_type) })
+
+        NArray[*na.transpose]
+      end
+
+      private
+
+      # @param ndvi [NArray]
+      # @return [NArray]
+      def calculate_ndvi_byte(ndvi)
+        ((ndvi + 1) * (255.0 / 2)).to_type(NArray::BYTE)
+      end
+
+      # @param ndvi [NArray]
+      # @return [NArray]
+      def calculate_ndvi_uint16(ndvi)
+        ((ndvi + 1) * (65_535.0 / 2)).to_type(NArray::INT)
+      end
```

Comment:
@tindron we don't need NDVI/etc stuff anymore, right? Maybe I can rip this out...

Reply:
Yeah, the NDVI stuff can go for sure.